rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -237,7 +237,17 @@ void SqliteStorage::create_topic(const rosbag2_storage::TopicMetadata & topic)
     insert_topic->bind(
       topic.name, topic.type, topic.serialization_format, topic.offered_qos_profiles);
     insert_topic->execute_and_reset();
-    topics_.emplace(topic.name, static_cast<int>(database_->get_last_insert_id()));
+
+    ROSBAG2_STORAGE_DEFAULT_PLUGINS_LOG_INFO("new topic %s", topic.name.c_str());
```

Comment:
Let's either remove these debug print statements, or format more informatively and bump down to `_DEBUG` severity.

Can you add a comment here as to why the logic is necessary? It's not obvious to me what the purpose of the SELECT statement is at a glance. For readability as well, it may be nice to write the result to a named variable e.g.

```
queried_topic_id = static_cast<int>(std::get<0>(*result)));
topics_.emplace(topic.name, queried_topic_id);
```

Reply:
Yes, I will remove the debug prints. They were there mostly to illustrate the issue. 
I am not sure if a db query is the right way to solve this. I will refactor it to be more readable if we decide to keep it.